src/Microsoft.AspNet.OData.Shared/Formatter/ODataOutputFormatterHelper.cs

```diff
@@ -158,7 +158,8 @@ internal static bool TryGetCharSet(MediaTypeHeaderValue mediaType, IEnumerable<s
             ODataMessageWriterSettings writerSettings = internalRequest.WriterSettings;
             writerSettings.BaseUri = baseAddress;
 
-            if (serializer.ODataPayloadKind == ODataPayloadKind.Delta)
+            // Set the response version to v4.01 when the request is a delta patch, but not any time we write a delta response payload.
+            if (serializer.ODataPayloadKind == ODataPayloadKind.Delta && internalRequest.Method == ODataRequestMethod.Patch)
```

Comment:
What's the internal request method of a normal delta response? Isn't it `Patch` also?

Reply:
or this is for other methods like GET, POST etc..

Reply:
That's 'Get', right?!

Reply:
PATCH is the verb used when the client is doing a delta (aka bulk) update.

The delta response payload is also used when the client retrieves changes using the deltaLink using GET.

Bulk updates are only supported for 4.01, so it's appropriate for the response from a delta update to be 4.01, but a request to get changes (which also returns in delta format) has been around since 4.0, and is valid in either 4.0 or 4.01 format.